portal/forms/teach_new.py

```diff
@@ -0,0 +1,519 @@
+        if (password or confirm_password) and password != confirm_password:
+            raise forms.ValidationError('Your new passwords do not match')
+
+        if not self.user.check_password(current_password):
+            raise forms.ValidationError('Your current password was incorrect')
+
+
+class TeacherLoginForm(forms.Form):
+    email = forms.EmailField(
+        label='Email address',
+        widget=forms.EmailInput(attrs={'placeholder': "my.email@address.com"}))
+    password = forms.CharField(
+        label='Password',
+        widget=forms.PasswordInput)
+
+    def clean(self):
+        if self.has_error('recaptcha'):
+            raise forms.ValidationError('Incorrect email address, password or captcha')
+
+        email = self.cleaned_data.get('email', None)
+        password = self.cleaned_data.get('password', None)
+
+        if email and password:
+
+            # Check it's a teacher and not a student using the same email address
+            user = None
+
+            user = self.find_user(email, user)
+
+            user = authenticate(username=user.username, password=password)
+
+            self.check_email_erros(user)
+
+            self.user = user
+
+        return self.cleaned_data
+
+    def find_user(self, email, user):
+        users = User.objects.filter(email=email)
+
+        for result in users:
+            if hasattr(result, 'userprofile') and hasattr(result.userprofile, 'teacher'):
+                user = result
+                break
+
+        if user is None:
+            raise forms.ValidationError('Incorrect email address or password')
+
+        return user
+
+    def check_email_erros(self, user):
+        if user is None:
+            raise forms.ValidationError('Incorrect email address or password')
+
+        if not user.is_active:
+            raise forms.ValidationError('User account has been deactivated')
+
+
+class ClassCreationForm(forms.Form):
+    classmate_choices = [('True', 'Yes'), ('False', 'No')]
+    class_name = forms.CharField(
+        label='Class Name',
+        widget=forms.TextInput(attrs={'placeholder': 'Lower KS2'}))
+    classmate_progress = forms.ChoiceField(
+        label="Allow students to see their classmates' progress?",
+        choices=classmate_choices,
+        widget=forms.Select(attrs={'class': 'wide'}))
+
+
+class ClassEditForm(forms.Form):
+    classmate_choices = [('True', 'Yes'), ('False', 'No')]
+    # select dropdown choices for potentially limiting time in which external students may join
+    # class
+    # 0 value = don't allow
+    # n value = allow for next n hours, n < 1000 hours
+    # o/w = allow forever
+    join_choices = [('', "Don't change my current setting"),
+                    ('0', "Don't allow external requests to this class"),
+                    ('1', "Allow external requests to this class for the next hour")]
+    for i in range(6):
+        hours = 4 * (i + 1)
```

Comment:
also, please note that this might be difficult to localise. For instance, Greek has a dual grammatical number so it will not be class vs classes but 1 class vs 2 classes vs 3 or more classes...